.github/workflows/release.yml

```diff
@@ -1,30 +1,39 @@
 name: Release
 
 on:
-  release:
-    types: [published]
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+  pull_request:
+    paths:
+      - .github/workflows/make_release.yml
 
 jobs:
   release:
     # requires that you have put your twine API key in your
     # github secrets (see readme for details)
     runs-on: ubuntu-latest
-    if: contains(github.ref, 'tags')
+    permissions:
+      contents: write
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: hynek/build-and-inspect-python-package@v2
+      - name: determine tag
+        run: echo "tag=${GITHUB_REF/refs\/tags\/v/}" >> "$GITHUB_ENV"
+      - name: Create Release
+        uses: "softprops/action-gh-release@v2"
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+        with:
+          tag_name: ${{ github.ref }}
+          name: ${{ env.tag }}
+          draft: false
+          prerelease: ${{ contains(env.tag, 'rc') || contains(env.tag, 'a') || contains(env.tag, 'b') }}
+          target_commitish: ${{ github.sha }}
```

Comment:
@LucaMarconato This line will save proper information in release to be later used by benchmarks.